waldo/add.go

```diff
@@ -104,35 +114,62 @@ func (aa *AddAction) confirmRecipe(basePath string, recipe *data.Recipe) bool {
 		appName = "(unknown)"
 	}
 
-	buildTool := recipe.BuildTool().String()
-	buildRoot := "(none)"
-	uploadToken := "(none)"
+	appID := recipe.AppID
+
+	if len(appID) == 0 {
+		appID = "(none)"
+	}
+
 	buildOptions := "(none)"
+	buildRoot := "(none)"
+	buildTool := recipe.BuildTool().String()
 
 	absPath := filepath.Join(basePath, recipe.BasePath)
 
 	if relPath := lib.MakeRelativeToCWD(absPath); len(relPath) > 0 {
 		buildRoot = relPath
 	}
 
-	if token := recipe.UploadToken; len(token) > 0 {
-		uploadToken = token
-	}
-
 	if summary := recipe.Summarize(); len(summary) > 0 {
 		buildOptions = summary
 	}
 
-	aa.ioStreams.Printf("  Platform:      %s\n", recipe.Platform)
 	aa.ioStreams.Printf("  App name:      %s\n", appName)
+	aa.ioStreams.Printf("  App ID:        %s\n", appID)
+	aa.ioStreams.Printf("  Platform:      %s\n", recipe.Platform)
 	aa.ioStreams.Printf("  Build tool:    %s\n", buildTool)
 	aa.ioStreams.Printf("  Build root:    %s\n", buildRoot)
-	aa.ioStreams.Printf("  Upload token:  %s\n", uploadToken)
 	aa.ioStreams.Printf("  Build options: %s\n", buildOptions)
 
 	return aa.promptReader.ReadYN(fmt.Sprintf("Add recipe %q", recipe.Name))
 }
 
+func (aa *AddAction) determineApp(platform lib.Platform) (string, string, error) {
+	if aa.options.Verbose {
+		aa.ioStreams.Printf("\nFetching %v apps for user token %q…\n", platform, aa.userToken)
+	}
+
+	items, err := api.FetchApps(aa.userToken, platform, aa.options.Verbose, aa.ioStreams)
+
+	if err != nil {
+		return "", "", err
+	}
+
+	if platform != lib.PlatformUnknown {
+		items = lib.CompactMap(items, func(item *tool.AppInfo) (*tool.AppInfo, bool) {
+			return item, item.Platform == platform
+		})
+	}
```

Comment:
Shouldn't this already be done by the backend if the query param is set in the request?

Reply:
Yes, but I believe in belt and suspenders.